File: soap-sdk/src/main/java/com/cleanster/soap/model/ApiResponse.java
package com.cleanster.soap.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

/** Generic API response wrapper (status + message). */
@JsonIgnoreProperties(ignoreUnknown = true)
public class ApiResponse {

    private int    status;
    private String message;

    public ApiResponse() {}

    public ApiResponse(int status, String message) {
        this.status  = status;
        this.message = message;
    }

    public int    getStatus()  { return status; }
    public String getMessage() { return message; }

    public void setStatus(int status)      { this.status = status; }
    public void setMessage(String message) { this.message = message; }

    public boolean isSuccess() {
        return status >= 200 && status < 300;
    }
}

File: android-sdk/src/main/kotlin/com/cleanster/android/model/ApiResponse.kt
package com.cleanster.android.model

import com.google.gson.annotations.SerializedName

data class ApiResponse<T>(
    @SerializedName("status")  val status: Int    = 0,
    @SerializedName("message") val message: String? = null,
    @SerializedName("data")    val data: T?        = null,
)
